### PR TITLE
argon: init at 2.0.21

### PR DESCRIPTION
**Pull request by StayBlue**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by FliegendeWurst — APPROVED** [review]

File: pkgs/by-name/ar/argon/package.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    [
+      udev
+      zstd
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      darwin.apple_sdk.frameworks.CoreFoundation
+      darwin.apple_sdk.frameworks.CoreServices
+      darwin.apple_sdk.frameworks.Foundation
+      darwin.apple_sdk.frameworks.Security
+      darwin.apple_sdk.frameworks.SystemConfiguration
```
Comment (FliegendeWurst) [inline pkgs/by-name/ar/argon/package.nix:36]:
```suggestion
```
Since my last review, the Darwin SDKs were updated to no longer require this, I think.

**Comment by StayBlue** [comment]
Sorry for the delay! I was busy over the past couple of days.

File: pkgs/by-name/ar/argon/package.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
+{ lib
+, rustPlatform
```
Comment (FliegendeWurst) [inline pkgs/by-name/ar/argon/package.nix:2]: Please use nixfmt-rfc-style

File: pkgs/by-name/ar/argon/package.nix (hunk 1 above)
Comment (StayBlue) [reply pkgs/by-name/ar/argon/package.nix]: The package didn't build on my M2 system without at least having Foundation and SystemConfiguration.

File: pkgs/by-name/ar/argon/package.nix (hunk 2 above)
Comment (StayBlue) [reply pkgs/by-name/ar/argon/package.nix]: Thanks for catching that!

File: pkgs/by-name/ar/argon/package.nix (hunk 1 above)
Comment (FliegendeWurst) [reply pkgs/by-name/ar/argon/package.nix]: You may need to rebase on master

**Review by FliegendeWurst — APPROVED** [review]

**Comment by FliegendeWurst** [comment]
Though I think it would be best to do a real rebase and drop the merge commit.

**Review by Aleksanaa — APPROVED** [review]

File: pkgs/by-name/ar/argon/package.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  darwin,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "argon";
+  version = "2.0.21";
+
+  src = fetchFromGitHub {
+    owner = "argon-rbx";
+    repo = "argon";
+    rev = version;
```
Comment (Scrumplex) [inline pkgs/by-name/ar/argon/package.nix:18]: nit
```suggestion
    rev = "refs/tags/${version}";
```

File: maintainers/maintainer-list.nix [hunk 4]
```diff
@@ -21101,6 +21101,12 @@
     github = "stasjok";
     githubId = 1353637;
   };
+  stayblue = {
```
Comment (Scrumplex) [inline maintainers/maintainer-list.nix:21104]: You can also use `StayBlue` instead of `stayblue` if you prefer

**Review by Scrumplex — APPROVED** [review]

File: pkgs/by-name/ar/argon/package.nix (hunk 3 above)
Comment (Aleksanaa) [reply pkgs/by-name/ar/argon/package.nix:18]: Or newish
```suggestion
    tag = version;
```

Comment (StayBlue) [reply pkgs/by-name/ar/argon/package.nix]: `tag` doesn't seem to work.

Comment (FliegendeWurst) [reply pkgs/by-name/ar/argon/package.nix]: > `tag` doesn't seem to work.

You would need to rebase on master first.

**Comment by Aleksanaa** [comment]
@ofborg build argon

File: pkgs/by-name/ar/argon/package.nix (hunk 3 above)
Comment (Aleksanaa) [reply pkgs/by-name/ar/argon/package.nix]: (Not a problem, don't push again)

**Comment by Scrumplex** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 351091`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>argon</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>argon</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>argon</li>
  </ul>
</details>

**Review by Scrumplex — APPROVED** [review]

**Pull request merged**